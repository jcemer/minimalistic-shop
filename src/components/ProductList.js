import React, { PropTypes } from 'react'

import ProductItem from './ProductItem'

const ProductList = ({ items, addToCart }) => {
  return (
    <div className="ui five column doubling cards">
      {items.map(item => <ProductItem key={item._id} addToCart={addToCart} {...item} />)}
    </div>
  )
}

ProductList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
  addToCart: PropTypes.func.isRequired
}

export default ProductList
