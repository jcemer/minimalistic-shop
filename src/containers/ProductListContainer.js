import React from 'react'
import { connect } from 'react-redux'

import { addToCart } from '../actions/cart'
import ProductList from '../components/ProductList'

export const ProductListContainer = ({ isReady, items, addToCart }) => {
  if (!isReady) return null

  return <ProductList addToCart={addToCart} items={items} />
}

export const mapStateToProps = (
  { products: { isReady, items } }
) => ({ isReady, items })

export default connect(mapStateToProps, { addToCart })(ProductListContainer)
