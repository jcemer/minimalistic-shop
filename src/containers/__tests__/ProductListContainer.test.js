import React from 'react'
import { shallow, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import { MemoryRouter } from 'react-router-dom'

import { ProductListContainer, mapStateToProps } from '../ProductListContainer'
import ProductList from '../../components/ProductList'

describe('ProductListContainer', () => {
  const addToCart = () => {}

  describe('<Component />', () => {
    it('should not render if it is not ready', () => {
      const wrapper = shallow(<ProductListContainer addToCart={addToCart} />)

      expect(wrapper.children().exists()).toBe(false)
    })

    it('should render ProductList with items', () => {
      const items = [{ _id: '123', image: 'url', price: '$1' }]
      const wrapper = shallow(<ProductListContainer addToCart={addToCart} isReady items={items} />)

      expect(wrapper.contains(<ProductList addToCart={addToCart} items={items} />)).toBe(true)
    })

    it('should render the proper HTML of ProductList with items', () => {
      const items = [{ _id: '123', image: 'url', price: '$1', stock: { remaining: 2 } }]
      const wrapper = render(
        <MemoryRouter>
          <ProductListContainer addToCart={addToCart} isReady items={items} />
        </MemoryRouter>
      )

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should returns no items if products is not ready', () => {
      const props = mapStateToProps(
        { products: { isReady: false } },
      )

      expect(props).toEqual({ isReady: false, items: undefined })
    })

    it('should returns items if products is ready', () => {
      const props = mapStateToProps(
        { products: { isReady: true, items: [] } }
      )

      expect(props).toEqual({ isReady: true, items: [] })
    })
  })
})
