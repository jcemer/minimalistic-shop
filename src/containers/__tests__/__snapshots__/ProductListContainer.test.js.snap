exports[`ProductListContainer <Component /> should render the proper HTML of ProductList with items 1`] = `
<div
  class="ui five column doubling cards">
  <div
    class="ui card">
    <a
      class="ui image"
      href="/product/123">
      <img
        alt="Product 123"
        src="url" />
    </a>
    <div
      class="content">
      <div>
        Price $1
      </div>
    </div>
    <div
      class="extra content">
      <button
        class="ui green button fluid">
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
